web/src/submessage.ts

```diff
@@ -7,32 +7,35 @@ import type {Message} from "./message_store";
 import type {Submessage} from "./types";
 import * as widgetize from "./widgetize";
 
-export const zform_widget_extra_data_schema = z.object({
-    choices: z.array(
-        z.object({
-            type: z.string(),
-            long_name: z.string(),
-            reply: z.string(),
-            short_name: z.string(),
-        }),
-    ),
-    heading: z.string(),
-    type: z.literal("choices"),
-});
-
-const poll_widget_extra_data_schema = z.object({
-    question: z.string().optional(),
-    options: z.array(z.string()).optional(),
-});
+export const zform_widget_extra_data_schema = z
+    .object({
+        choices: z.array(
+            z.object({
+                type: z.string(),
+                long_name: z.string(),
+                reply: z.string(),
+                short_name: z.string(),
+            }),
+        ),
+        heading: z.string(),
+        type: z.literal("choices"),
+    })
+    .nullable();
```

Comment:
Can the `zform_widget_extra_data` ever be `null` ( `.nullable()` )?  The schema I created 
```
            .union([zform_widget_extra_data_schema, poll_widget_extra_data_schema])
            .nullable(),
```
had `.nullable()` to account for the case when `widget_type` is `todo` (as the`extra_data` field used to be `null` before this work)